File: data/src/main/java/com/clloret/days/data/local/entities/DbTag.java
package com.clloret.days.data.local.entities;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "tags")
public class DbTag {

  @PrimaryKey
  @ColumnInfo(name = "id")
  @NonNull
  private String id;

  @ColumnInfo(name = "name")
  private String name;

  public DbTag(@NonNull String id, String name) {

    this.id = id;
    this.name = name;
  }

  @Ignore
  public DbTag(@NonNull String id) {

    this.id = id;
  }

  @NonNull
  public String getId() {

    return id;
  }

  public void setId(@NonNull String id) {

    this.id = id;
  }

  public String getName() {

    return name;
  }

  public void setName(String name) {

    this.name = name;
  }
}

File: data/src/main/java/com/clloret/days/data/local/entities/DbTag.kt
package com.clloret.days.data.local.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = "tags")
class DbTag {
  @PrimaryKey
  @ColumnInfo(name = "id")
  var id: String

  @ColumnInfo(name = "name", defaultValue = "")
  var name: String = ""

  constructor(id: String, name: String) {
    this.id = id
    this.name = name
  }

  @Ignore
  constructor(id: String) {
    this.id = id
  }

}